src/com/xilinx/rapidwright/edif/EDIFParser.java

```diff
@@ -672,17 +618,44 @@ private EDIFPortInst parseEDIFPortInst(){
 		}
 		
 		currToken = getNextToken();
+		EDIFCell portCell = parentCell;
 		if(currToken.equals(LEFT_PAREN)){
 			expect(INSTANCEREF,getNextToken());
 			String instanceref = getNextToken();
-			portInst.setCellInst(getRefEDIFCellInst(instanceref));
+			portInst.setCellInstRaw(getRefEDIFCellInst(instanceref));
+			portCell = portInst.getCellInst().getCellType();
 			expect(RIGHT_PAREN,getNextToken());
 			expect(RIGHT_PAREN,getNextToken());
 		}else{
 			// This is a port to higher level
 			expect(RIGHT_PAREN,currToken);
 		}
+		EDIFPort port = portCell.getPort(portInst.getName());
+		if(port == null) {
+		    // Finding by EDIFName is O(n), but n is generally small and alternative to building
+		    // a map for this single search ends up taking longer 
+		    for(Entry<String,EDIFPort> e : portCell.getPortMap().entrySet()) {
+		        if(e.getValue().getLegalEDIFName().equals(portInst.getName())) {
+		            port = e.getValue();
+		            break;
+		        }
+		    }
+		}
 
+		if(port == null) {
+		    throw new RuntimeException("ERROR: Couldn't find EDIFPort for "
+		            + "EDIFPortInst " + portInst.getName() + " on line number " + lineNumber);
+		}
+		portInst.setPort(port);
+		String portInstName = portInst.getPortInstNameFromPort();
+		String dedupName = stringPool.putIfAbsent(portInstName, portInstName); 
+		if(dedupName == null) {
+		    dedupName = portInstName;
+		}
```

Comment:
Oops sorry. I think I meant to ask for `computeIfAbsent` which, at the cost of a lambda, returns the old or new value so you can do:
```suggestion
		String dedupName = stringPool.computeIfAbsent(portInstName, (k) -> portInstName); 
```

Reply:
Wow, that's a big difference to get rid of `get`/`put`! I like `computeIfAbsent` for its one liner aspect, rather than the trivial lambda that needs to be supplied. Are these benchmarks for changing just this usage here, or did you change it in any more places? I've noticed the get/put pattern in some other places too.

Reply:
There were three instances in the parser, I changed all three.  Might not have as big of an impact in all other scenarios, though.  Should we merge this into #339?

Reply:
> There were three instances in the parser, I changed all three. Might not have as big of an impact in all other scenarios, though. Should we merge this into #339?

No harm in doing so as they're so inter-related.